apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientResourceProvider.java

```diff
@@ -421,11 +433,41 @@ private Beneficiary selectBeneWithLatestReferenceYear(List<Beneficiary> duplicat
 
     static IncludeIdentifiersMode determineIncludeIdentifiersMode(RequestDetails requestDetails) {
       String includeIdentifiersValue = requestDetails.getHeader(HEADER_NAME_INCLUDE_IDENTIFIERS);
-      if (Boolean.parseBoolean(includeIdentifiersValue) == true) {
+
+      List<String> headerValues = Arrays.asList(includeIdentifiersValue.split("\\s*,\\s*"));
+
+      if (includeIdentifiersValue == null) {
+        // If the header was not included in the request or blank
+        return OMIT_HICNS_AND_MBIS;
+        // Still match boolean true by ignoring case
+      } else if (headerValues.stream().anyMatch(x -> x.equalsIgnoreCase("true"))) {
+        return INCLUDE_MBIS;
+      } else if (headerValues.contains("hicn") && headerValues.contains("mbi")) {
         return INCLUDE_HICNS_AND_MBIS;
+      } else if (headerValues.contains("hicn")) {
+        return INCLUDE_HICNS;
+      } else if (headerValues.contains("mbi")) {
+        return INCLUDE_MBIS;
       } else {
+        // Default
         return OMIT_HICNS_AND_MBIS;
       }
     }
+
+    /** @return Returns true if the @param includeIdentifiersMode includes unhashed hicn */
+    public static boolean hasHICN(IncludeIdentifiersMode includeIdentifiersMode) {
+      if ((includeIdentifiersMode == IncludeIdentifiersMode.INCLUDE_HICNS_AND_MBIS)
+          || (includeIdentifiersMode == IncludeIdentifiersMode.INCLUDE_HICNS)) {
+        return true;
+      } else return false;
+    }
+
+    /** @return Returns true if the @param includeIdentifiersMode includes unhashed mbi */
+    public static boolean hasMBI(IncludeIdentifiersMode includeIdentifiersMode) {
+      if ((includeIdentifiersMode == IncludeIdentifiersMode.INCLUDE_HICNS_AND_MBIS)
```

Comment:
similar to the hasHicn try reworking to `return headerValues.contains("mbi")`

Reply:
I'm going to try reworking this by removing the `IncludeIdentifiersMode` and utilizing a `headerValues` List instead (will probably rename this). Was originally worried about changing the original code too much. Will update soon. TY!